projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -355,6 +356,38 @@ function wpcom_get_domain_upsell_badge_text() {
 	return wpcom_is_checklist_task_complete( 'domain_upsell' ) ? '' : __( 'Upgrade plan', 'jetpack-mu-wpcom' );
 }
 
+/**
+ * Determines whether or not domain upsell task should be visible.
+ *
+ * @return bool True if user is on a free plan with no custom domain name.
+ */
+function wpcom_is_domain_upsell_visible() {
```

Comment:
Nit: it may be cleaner to call this something like `wpcom_is_domain_upsell_task_visible()`, as we have lots of domain upsells, so it would be better to avoid any possible confusion or ambiguity.
```suggestion
function wpcom_is_domain_upsell_task_visible() {
```